hiyapyco/__init__.py

```diff
@@ -416,7 +428,7 @@ def _deepmerge(self, a, b):
             pass
         if a is None or isinstance(b, primitiveTypes):
             logger.debug('deepmerge: replace a "%s"  w/ b "%s"' % (a, b,))
-            a = b
+            a = self.mergeprimitive(a, b, context)
```

Comment:
see comment above.
I know it is not that fancy and `pythonic`, but it increases readability IMHO
```diff
if a is None or isinstance(b, primitiveTypes):
-            logger.debug('deepmerge: replace a "%s"  w/ b "%s"' % (a, b,))
-            a = self.mergeprimitive(a, b, context)
+            if self.mergeprimitive is None:
+                logger.debug('deepmerge: replace a "%s"  w/ b "%s"' % (a, b,))
+                a = b
+            else:
+                logger.debug('deepmerge: call mergeprimitive for a "%s",  b "%s" with context' % (a, b,))
+                a = self.mergeprimitive(a, b, context)
         elif isinstance(a, listTypes):
```

Reply:
Hello @malachib 
Please include the diff above in a new commit
Thx